upload/src/addons/SV/WarningImprovements/XF/Service/User/ContentChange.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+namespace SV\WarningImprovements\XF\Service\User;
+
+/**
+ * Class ContentChange
+ *
+ * @package SV\WarningImprovements
+ */
+class ContentChange extends XFCP_ContentChange
+{
+    /**
+     * ContentChange constructor.
+     *
+     * @param \XF\App $app
+     * @param         $originalUserId
+     * @param null    $originalUserName
+     */
+    public function __construct(\XF\App $app, $originalUserId, $originalUserName = null)
+    {
+        parent::__construct($app, $originalUserId, $originalUserName);
+        if ($this->newUserId !== 0) // deleting user
+        {
+            $this->steps[] = 'stepReassignWarningActions';
+        }
+    }
+
+    /**
+     * @throws \XF\PrintableException
+     */
+    protected function stepReassignWarningActions()
+    {
+        /** @var \SV\WarningImprovements\XF\Repository\UserChangeTemp $userChangeTempRepo */
+        $userChangeTempRepo = $this->repository('XF:UserChangeTemp');
+        /** @var \SV\WarningImprovements\XF\Entity\User $targetUser */
+        $targetUser = $this->em()->find('XF:User', $this->newUserId);
+
+        $warningActionsAppliedToSource = $userChangeTempRepo->getWarningActions($this->originalUserId, true)->fetch();
+        if ($warningActionsAppliedToSource->count())
+        {
+            $warningActionIds = [];
+            /** @var \XF\Entity\UserChangeTemp|\SV\WarningImprovements\XF\Entity\UserChangeTemp $_warningAction */
+            foreach ($warningActionsAppliedToSource AS $_warningAction)
+            {
+                $warningActionDetails = explode('_', $_warningAction->change_key);
+                if (!isset($warningActionDetails[2]))
```

Comment:
Should check that `$warningActionDetails[0] == 'warning'` & `$warningActionDetails[1] == 'action'` (IIRC that is the change key used by warning actions)

Reply:
`getWarningActions()` adds a where condition `$warningActions->where('change_key', 'LIKE', 'warning_action_%');` which should show only `warning_action` logs